SkyRocketToys/ardupilot
# Copter: Improve setting home location when it takes off in indoor mode

To test this PR with a real copter:

1.Ready two PC/Phone to connect to the web server on the drone. I used two phones for my test- one connects to the web server using google chrome to see the status text and one connects with the skyviper video viewer app to check where it sets the home location.
2.connect the battery to drone and turn on tx
3.on you pc/phone, connect the 1st PC/Phone to the web server on the drone via wifi.
4.using your any web browser (I used chrome) navigate to 192.168.99.1
5.navigate to system status>>message
6.arm or arm and hover the drone in alt_hold mode
7.keep an eye on the message tab - you should get the "INDOOR MODE:set home loc with horizontal_acc:" message
8.fly forward or in any direction which is considerable far from you i.e. 30meter away before it gets full gps lock/EKF happy
9. wait for the drone to get full gps lock/ EKF happy
10. connect your 2nd PC/Phone to the drone and launch either mission planner if you are using PC or sky viper video viewer app if you are using a phone.
11. check if the home location is near you and make sure it did not set home location in some part of the world.
12. if home is nearby then RTL, if not manually control the drone and land it safely because something is wrong

I have attached my test video and some logs on that flight for your reference
https://drive.google.com/open?id=0B3_WlAMsda15R25KTl96ejJXbDQ

To test this PR with a SITL:

1. I did sim_vehicle.py --console --out=udp:host:port to connect to my windows machine running mission planner
2. param set sim_gps_numsats 2
3. restart sitl
4.start mission planner on windows
5.switch to alt_hold mode
6.arm and hover:
7.you will see the status text message  "INDOOR MODE:set home loc with horizontal_acc" - you should see home location get set on mission planner 
8.fly the copter just a couple of meters away from where it launched
9.do param set sim_gps_numsats 12
10.wait for it to get full lock/ekf happy
11. do RTL
12. check if it goes back to the desired home location. if not then something is wrong
